src/Module.h

```diff
@@ -16,6 +16,14 @@
 #include "Target.h"
 
 namespace Halide {
+
+/** Type of linkage a function can have. */
```

Comment:
I think now that it's outside of LoweredFunc the term "function" needs elaboration in the comment. How about "Type of linkage a function in a lowered Halide module can have. Also controls whether auxiliary functions and metadata are generated. "?